src/adaptors/hyperdrive/index.js

```diff
@@ -0,0 +1,324 @@
+const ethers = require("ethers")
+const axios = require('axios');
+const sdk = require('@defillama/sdk');
+const utils = require('../utils');
+const providers = require('@defillama/sdk/build/providers.json');
+const { GET_POOL_CONFIG_ABI, GET_POOL_INFO_ABI, POSITION_ABI, MARKET_ABI } = require('./abi');
+
+const FUTURE_REGISTRY_ADDRESS = "0x6668310631Ad5a5ac92dC9549353a5BaaE16C666"
+
+const config = {
+  ethereum: { registry: '0xbe082293b646cb619a638d29e8eff7cf2f46aa3a', },
+  xdai: { registry: '0x666fa9ef9bca174a042c4c306b23ba8ee0c59666', },
+  base: {},
+  linea: {},
+}
+
+async function queryPoolHoldings(poolContract, config, name) {
+  let baseTokenBalance;
+  let vaultSharesBalance;
+  let vaultContractAddress;
+
+  // Query base token balance
+  if (config.baseToken === '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE') {
+    // ETH case
+    baseTokenBalance = (
+      await sdk.api.eth.getBalance({
+        target: poolContract.address,
+        chain: config.chain
+      })
+    ).output;
+  } else if (name?.includes(' LP ')) {
```

Comment:
idea was to so it doesn't match names that have "LP" in the name otherwise, like "BALLPIT YIELD SOURCE", but switched to matching on kind (still not perfect) in commit ddfbe5d712eeb3167eab9b9718d482e452c78dc5